### PR TITLE
Story Cards 3,4,11 & 16

### PR DESCRIPTION
**Pull request by KamilRosiak**
Added new functions:
1. Now you have a Search Window in the ConstraintView to search through the Constraints (supports regex search)
2. You can delete, edit and create Constraints plus select and delete more than one feature at a time in the ConstraintView.
3. The ConstraintView adapts when you select Features in the Feature Diagram (show only Constraints that contains the Feature(s) selected), also reacts on "show collapsed Constraints" - function
4. When you switch between the pages of a Feature Model (e.g. look at the Source Code instead of the Feature Diagram) the ContextView shows "Open a feature diagram" instead of showing all the Constraints.

### CODE REVIEW

**Review by ThisIsATunaSandwich — COMMENTED** [review]
Looking good!!!
@PaulWestphal

**Comment by PaulWestphal** [comment]
"Edit constraint" should not be available when no constraint or more than one constraint is selected. Looks good.

**Comment by PaulWestphal** [comment]
The most recent pull request hasn't changed the availability of "Edit Constraint". Did you see my comment? @ksmallros

File: plugins/de.ovgu.featureide.fm.core/src/de/ovgu/featureide/fm/core/localization/StringTable.java [hunk 1]
```diff
@@ -617,6 +617,7 @@
 	public static final String BUILD_PATH_RESTRICTION_ANDROID = PATH_RESTRICTION_ANDROID + BUILD_PATH + "\".";
 	public static final String CONFIG_PATH_RESTRICTION_ANDROID = PATH_RESTRICTION_ANDROID + CONFIG_PATH + "\".";
 	public static final String OPEN_A_FEATURE_MODEL_ = "Open a feature model.";
+	public static final String OPEN_A_FEATURE_DIAGRAM_EDITOR = "Open a feature diagram editor.";
```
Comment (tthuem) [inline plugins/de.ovgu.featureide.fm.core/src/de/ovgu/featureide/fm/core/localization/StringTable.java:620]: Please change to "Open a feature diagram." for consistency

File: plugins/de.ovgu.featureide.fm.ui/src/de/ovgu/featureide/fm/ui/editors/IGraphicalFeatureModel.java [hunk 2]
```diff
@@ -101,6 +101,8 @@
 	 *
 	 * @return All not collapsed constraints stored in this feature model.
 	 */
+	List<IGraphicalConstraint> getNonCollapsedConstraints();
+
 	List<IGraphicalConstraint> getVisibleConstraints();
```
Comment (tthuem) [inline plugins/de.ovgu.featureide.fm.ui/src/de/ovgu/featureide/fm/ui/editors/IGraphicalFeatureModel.java:106]: What is the difference between those two methods? Would be good to have a comment here.

File: plugins/de.ovgu.featureide.fm.ui/src/de/ovgu/featureide/fm/ui/editors/featuremodel/operations/ExpandConstraintOperation.java [hunk 3]
```diff
@@ -45,7 +45,7 @@
 	private final LinkedList<IGraphicalFeature> affectedFeatureList = new LinkedList<IGraphicalFeature>();
 
 	/**
-	 * @param featureModel
+	 * @param currentModel
 	 * @param label
```
Comment (tthuem) [inline plugins/de.ovgu.featureide.fm.ui/src/de/ovgu/featureide/fm/ui/editors/featuremodel/operations/ExpandConstraintOperation.java:49]: The whole comment block does not make sense without any data and this change introducing another wrong variable name does not help either.

File: plugins/de.ovgu.featureide.fm.ui/src/de/ovgu/featureide/fm/ui/editors/ConstraintDialog.java [hunk 4]
```diff
@@ -539,7 +539,7 @@ private void cancelButtonPressEvent() {
 	/**
 	 * closes the shell and adds new constraint to the feature model if possible
 	 *
-	 * @param featureModel
+	 * @param currentModel
 	 * @param constraint
 	 */
```
Comment (tthuem) [inline plugins/de.ovgu.featureide.fm.ui/src/de/ovgu/featureide/fm/ui/editors/ConstraintDialog.java:544]: The method does not have any parameters! Changing the name of a non-existent parameter does not seem useful.

**Pull request merged**